pkg/controller/node/nodecontroller.go

```diff
@@ -79,6 +79,16 @@ var (
 		Key:    metav1.TaintNodeNotReady,
 		Effect: v1.TaintEffectNoExecute,
 	}
+
+	// TODO(resouer) move this to metav1 well_known_labels.go after #42406 get merged.
+	// TaintNodeNoSchedule would be automatically added by node controller when
+	// node runs into a unschedulable condition and removed when node becomes schedulable.
+	TaintNodeNoSchedule = "node.beta.kubernetes.io/noSchedule"
```

Comment:
Yes, I think that was the plan. And I know it's pretty scary, but even more scary would be enabling it by default only after it goes to GA, when we're generally stuck with it. This is why we need extensive testing when something as big as that goes to beta.